domain/machine/watcher_test.go

```diff
@@ -136,6 +136,141 @@ func (s *watcherSuite) TestMachineCloudInstanceWatchWithDelete(c *gc.C) {
 	watcherC.AssertChange(machineUUID)
 }
 
+// TestMachineRebootWatchNoRebootRequired tests the functionality of watching for machine reboot
+// events when no machine has requested a reboot yet. It creates a watcher, asserts that there are no
+// initial events indicating reboot requests, and asserts that the change stream is idle.
+func (s *watcherSuite) TestMachineRebootWatchNoRebootRequired(c *gc.C) {
+	watcher, err := s.svc.WatchMachineReboot(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	defer watchertest.CleanKill(c, watcher)
+
+	watcherC := watchertest.NewStringsWatcherC(c, watcher)
+	// Initial event.
+	watcherC.AssertChange() // No machine asked for reboot yet
+	s.AssertChangeStreamIdle(c)
```

Comment:
Can you use the watcher test harness. See https://github.com/juju/juju/blob/6ba4623e248dfb2c8f630fd70497802291fa1716/domain/application/watcher_test.go#L31-L75 as an example.

Reply:
It is already the case, this harness is loaded in the test suite in the `svc` field during setup:

https://github.com/juju/juju/pull/17925/files/8394d0576a6f075529048fee80f8e732314466b3#diff-62a057f1a0a662825d70f7ece1277da7e1cd569c6ab804b41058f2d7c3ad3679R32-R43

Reply:
I mean the watcher test harness

Reply:
I tried to play with that harness but I am not a big fan of it because it makes failing test more difficult to analyze, since the trace will generate two reference in the code:
   *  One to the line where harness.Run() is called
   * Another to the assertion definition in watchertest package. 
   
Neither of them are useful for understanding what is going on. I think tests should be written to fail, which meant that we don't really care how they  behave when they pass, but we really want to understand them when they fail.

So, in case of failure, we should be directed to the exact assertion which fails, with the most context we can get (actual vs expected data, and some insight of what could have cause the regression, if conceivable whenever the test were written.) We don't really care if the test are not really DRY, as long as they answer to this purpose. 

Actual implementation of this harness fails to answer to this purpose, so I am not a big fan of it, and I don't have enough insight to see how it improves the test right now.

**side note**: I am suspicious of the correctness of the harness since I replaced some check in `application/watcher_test.go` by obvious failing one, like below:

```go
	var id corecharm.ID
	harness.AddTest(func(c *gc.C) {
		id, _, err = svc.SetCharm(context.Background(), charm.SetCharmArgs{
			Charm:    &stubCharm{},
			Source:   internalcharm.CharmHub,
			Revision: 1,
		})
		c.Assert(err, jc.ErrorIsNil)
	}, func(w watchertest.WatcherC[[]string]) {
		w.Check(
			watchertest.StringSliceAssert[string]("failure"), // force failure, just because...
		)
	})
```
 ... And it doesn't fail. I may have miss something because it seems too big to not be my mistake, but... well, it make me suspicious.